### PR TITLE
Update Dockerfile adding back in `public` directory

### PR DESCRIPTION
**Pull request by dmsnell**
See #8516

In the build I mistakenly prevented copying in the `public` directory
thinking that it was generated during the build process.

Actually it held our static assets and needed to be there. Images were
missing, for example.

The same Dockerfile has been updated again but this time I have not
prevented Docker from incorporating the `public` directory

Further I have added back the execution of `/tmp/env-config.sh` which I somehow accidentally removed before.

**Testing**
You can build the Docker image and run it locally. Yesterday it all seemed to deploy fine at least, so we have a certain level of confidence that this will go smoothly.

**WRONG: Omitting `public`**
screen shot 2016-10-07 at 12 04 59 pm

**Right: _not_ omitting `public`**
screen shot 2016-10-07 at 12 25 23 pm

cc: @gwwar @alisterscott @umurkontaci @nylen

### CODE REVIEW

**Comment by gwwar** [comment]
👍 Tested this. Image builds fine, and drake is available from the public folder. Let's wait until Monday to 🚢 though

**Comment by umurkontaci** [comment]
I'm not sure how the build system works, but I'd like to add `RUN touch ./env-config.sh` as the first command to the series of commands.

This way we can remove the need to create an empty file in local repo to trigger the build.

If the `/env-config.sh` is mounted, we can probably do something like `RUN test -f ./env-config.sh || touch ./env-config.sh`.

**Comment by umurkontaci** [comment]
Also, another question slightly out of the scope of this PR: Why do build production bundle twice?

**Comment by dmsnell** [comment]
> Why do build production bundle twice?

I'm not sure what you're referring to with _twice_ here. Currently the image builds all four production environments and then stores the built files in the image. Then when deployed it will spool up the appropriate version of Calypso.

On a related note I don't think we should be building all four environments on every build when only one is required. We could probably split a single build into four builds as separate images: `wp-calypso:wpcalypso`, `wp-calypso:production`, `wp-calypso:horizon`, `wpcalypso:stage`.

There is significant room for improvement here. I am excited to move in that direction. My personal target for the near future is to be able to build the image locally in under three minutes and correspondingly decrease the time it takes to deploy.

**Comment by umurkontaci** [comment]
> I'm not sure what you're referring to with twice here. Currently the image builds all four production environments and then stores the built files in the image. Then when deployed it will spool up the appropriate version of Calypso.

👍 


> On a related note I don't think we should be building all four environments on every build when only one is required.

Definitely agree with this.

File: Dockerfile [hunk 1]
```diff
@@ -1,49 +1,64 @@
...
-          rm node-v$NODE_VERSION-linux-x64.tar.gz
+WORKDIR    /calypso
 
-# npmrc is created by env-config.sh. For local testing, an empty one is generated
-RUN     touch /usr/local/etc/npmrc && \
-          mkdir /usr/local/node/etc && \
-          cp /usr/local/etc/npmrc /usr/local/node/etc/npmrc
 
-ENV     NODE_PATH /calypso/server:/calypso/client
+ENV        NODE_PATH=/calypso/server:/calypso/client
```
Comment (gwwar) [inline Dockerfile]: For anyone reading the `KEY=VALUE` syntax can be used to set multiple env variables. https://docs.docker.com/engine/reference/builder/#env

**Comment by gwwar** [comment]
Retested this and the docker image works well. Feel free to 🚢  but let's make sure the queue is clear when you merge this.

**Comment by dmsnell** [comment]
thanks @gwwar for the review - I made another update - mind checking it out?

here's the results of my timing tests (on machine):
 - if we start from scratch: 8-9 minutes
 - if we change a dependency: 8-9 minutes
 - if we only change Calypso code:  6-6.5 minutes

important is the last one, which is four times as high as it would need to be if we only built a single environment. this roughly means that if we're primed on the build and dependencies don't change, building a single environment should take around 1.5 minutes (should equal the time it takes to build Calypso)

**Comment by blowery** [comment]
> important is the last one, which is four times as high as it would need to be if we only built a single environment. this roughly means that if we're primed on the build and dependencies don't change, building a single environment should take around 1.5 minutes (should equal the time it takes to build Calypso)

So would we build a separate image for each deployment target, likely in parallel?

**Comment by dmsnell** [comment]
> So would we build a separate image for each deployment target, likely in parallel?

for local dev purposes we should be able to get by with only building one. this should be able to happen without impacting the production build

**Pull request merged**